### PR TITLE
chore: remove the remaining llvm references from cxx/common (#945)

### PR DESCRIPTION
**Pull request by zrlk**

### CODE REVIEW

File: kythe/cxx/common/kythe_metadata_file.cc [hunk 1]
```diff
@@ -210,9 +212,9 @@ bool KytheMetadataSupport::LoadMetaElement(const rapidjson::Value& value,
 #undef JSON_SAFE_LOAD_STRING
 
 std::unique_ptr<MetadataFile> KytheMetadataSupport::LoadFromJSON(
-    llvm::StringRef json) {
+    absl::string_view json) {
   rapidjson::Document document;
-  document.Parse(json.str().c_str());
+  document.Parse(std::string(json).c_str());
```
Comment (shahms) [inline kythe/cxx/common/kythe_metadata_file.cc]: Does this accept (const char*, size) as well? Or just null-terminated string?

Comment (shahms) [reply kythe/cxx/common/kythe_metadata_file.cc]: https://github.com/Tencent/rapidjson/blob/master/include/rapidjson/document.h#L2342

It does! we can save this copy and conversion to string.

Comment (zrlk) [reply kythe/cxx/common/kythe_metadata_file.cc]: rapidjson has its own custom collection of input streams we could enjoy if we choose.

Comment (zrlk) [reply kythe/cxx/common/kythe_metadata_file.cc]: Oh, they must build their docs without `RAPIDJSON_HAS_STDSTRING` set

Comment (shahms) [reply kythe/cxx/common/kythe_metadata_file.cc]: That I can't say, but there is a `Parse(const char*, size_t)` overload, which should be directly compatible with `json.data(), json.size()`

**Pull request merged**